client/state/data-layer/wpcom/jetpack/settings/index.js

```diff
@@ -149,7 +130,7 @@ export const retryOrAnnounceSaveFailure = ( { dispatch }, action, { message: err
 	// since it might just be a slow server that actually ends up installing it
 	// properly, in which case a subsequent request will return 'success'.
 	if (
-		get( settings, 'installWooCommerce' ) !== true ||
+		get( settings, [ 'onboarding', 'installWooCommerce' ] ) !== true ||
```

Comment:
If we keep those handlers generic, I think this is a pretty specific functionality to be defined in such a location. Could it make more sense to move it in some sort of middleware (not necessarily in this PR)?

Reply:
Should be done on this PR since it's where it's introduced. 👉11b3121

Reply:
Yep, this is more specific than anything in this file should be. I just haven't really figured out how to elegantly solve this otherwise. A middleware could be an option, I'm just not too seasoned in writing those, and I hope that middleware on top of (data-layer) middleware won't cause any problems. Anyway, warrants a separate PR IMO.